### PR TITLE
Feature: Open a Dialog on the side

### PR DESCRIPTION
**Pull request by MPapst**
superseeds #690

As discussed this implementation is based on the `DialogService` and the `RadzenDialog` component.

### CODE REVIEW

File: Radzen.Blazor/RadzenDialog.razor [hunk 1]
```diff
@@ -8,11 +8,51 @@
     <DialogContainer Dialog=@dialog ShowMask=@(dialog==dialogs.LastOrDefault()) />
 }
 
+@if (isSideDialogOpen)
+{
+    <aside
+        class="@($"rz-dialog-side {sideDialogOptions.CssClass}")"
+        tabindex="@(isSideDialogOpen ? "0" : "-1")"
+        style="@($"{sideDialogOptions.Style}{(sideDialogOptions.Width != null ? " width: {sideDialogOptions.Witdth}" : "")}")"
```
Comment (MPapst) [inline Radzen.Blazor/RadzenDialog.razor:16]: Style with optional set Width, which overrides the "hard coded" 400px width in the css

**Comment by MPapst** [comment]
Hi @enchev,

I'll have a look on the `DialogPosition`. Do you think bottom and top makes sense in this case?
Style is already implemented as part of the DialogOptions, see comment above.

**Comment by enchev** [comment]
Thanks @MPapst! In our opinion Top and Bottom most probably will be requested immediately if we merge it with just Left|Right.

**Comment by MPapst** [comment]
Hmm, will try to add them later today.
Those imply a height property over a width... that will bring a way more complex implementation.

**Comment by MPapst** [comment]
@enchev if you don't mind, I fixed the Nullreference Exception introduced with the format replacement in RadzenNumeric

**Pull request merged**